src/main/java/it/gov/pagopa/pu/debtpositions/service/AuthorizeOperatorOnDebtPositionTypeServiceImpl.java

```diff
@@ -1,27 +1,47 @@
 import it.gov.pagopa.pu.debtpositions.model.DebtPositionTypeOrg;
+import it.gov.pagopa.pu.debtpositions.model.DebtPositionTypeOrgOperators;
+import it.gov.pagopa.pu.debtpositions.repository.DebtPositionTypeOrgOperatorsRepository;
 import it.gov.pagopa.pu.debtpositions.repository.DebtPositionTypeOrgRepository;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 
+import static it.gov.pagopa.pu.debtpositions.util.SecurityUtils.SYSTEM_USERID_PREFIX;
+
 @Service
 public class AuthorizeOperatorOnDebtPositionTypeServiceImpl implements AuthorizeOperatorOnDebtPositionTypeService {
 
-    private final DebtPositionTypeOrgRepository debtPositionTypeOrgRepository;
+  private static final String ORG_USER_ID_PREFIX = "WS_USER-";
 
-    public AuthorizeOperatorOnDebtPositionTypeServiceImpl(DebtPositionTypeOrgRepository debtPositionTypeOrgRepository) {
-      this.debtPositionTypeOrgRepository = debtPositionTypeOrgRepository;
-    }
+  private final DebtPositionTypeOrgRepository debtPositionTypeOrgRepository;
+  private final DebtPositionTypeOrgOperatorsRepository debtPositionTypeOrgOperatorsRepository;
+
+  public AuthorizeOperatorOnDebtPositionTypeServiceImpl(DebtPositionTypeOrgRepository debtPositionTypeOrgRepository, DebtPositionTypeOrgOperatorsRepository debtPositionTypeOrgOperatorsRepository) {
+    this.debtPositionTypeOrgRepository = debtPositionTypeOrgRepository;
+    this.debtPositionTypeOrgOperatorsRepository = debtPositionTypeOrgOperatorsRepository;
+  }
 
-  public DebtPositionTypeOrg authorize(Long debtPositionTypeOrgId, String operatorExternalUserId) {
-    Optional<DebtPositionTypeOrg> debtPositionTypeOrg =
-      debtPositionTypeOrgRepository.findByDebtPositionTypeOrgIdAndOperatorExternalUserId(debtPositionTypeOrgId, operatorExternalUserId);
+  public DebtPositionTypeOrg authorize(String orgIpaCode, Long debtPositionTypeOrgId, String operatorExternalUserId) {
+    DebtPositionTypeOrg debtPositionTypeOrg = debtPositionTypeOrgRepository.findById(debtPositionTypeOrgId)
+      .orElseThrow(() -> new NotFoundException("The DebtPositionTypeOrg with id " + debtPositionTypeOrgId + " was not found"));
 
-    return debtPositionTypeOrg
-      .orElseThrow(() -> new OperatorNotAuthorizedException("The operator " + operatorExternalUserId + " is not authorized on the DebtPositionTypeOrg " + debtPositionTypeOrgId));
+    if (!isOperatorAuthorized(orgIpaCode, debtPositionTypeOrgId, operatorExternalUserId)) {
+      throw new OperatorNotAuthorizedException("The operator " + operatorExternalUserId + " is not authorized on the DebtPositionTypeOrg " + debtPositionTypeOrgId);
+    }
+
+    return debtPositionTypeOrg;
+  }
 
+  private boolean isOperatorAuthorized(String orgIpaCode, Long debtPositionTypeOrgId, String operatorExternalUserId) {
+    if (operatorExternalUserId.startsWith(SYSTEM_USERID_PREFIX) || operatorExternalUserId.equals(ORG_USER_ID_PREFIX + orgIpaCode)) {
```

Comment:
SIL client starts with the configured String, not equals:
see SIL users documented in: https://pagopa.atlassian.net/wiki/spaces/SPAC/pages/1405845916/Architettura+microservizi#3.1.-Classi-di-utenze
```suggestion
    if (operatorExternalUserId.startsWith(SYSTEM_USERID_PREFIX) || operatorExternalUserId.startsWith(ORG_USER_ID_PREFIX + orgIpaCode)) {
```